### DISCUSSION TITLE
How do I print a complete configuration file?

### DISCUSSION THREAD

**Discussion by IncludeMathH** [open-mmlab/mmocr#1545] [Q&A] [answered]
In previous versions of mmocr, as well as in projects such as mmdetection, there was a program (tools/misc/print_config.py) that could print out the complete configuration file directly, but the current 1.x version could not find this useful tool. May I ask how you have solved the problem?

**Comment by gaotongxiao** [comment, marked as answer]
That's a nice catch. It was accidentally removed but we can add it back to `tools/anaylsis_tools`. There is one reference in [MMDetection](https://github.com/open-mmlab/mmdetection/blob/08c42c03d7c45b2b9c9b83bffa0ad08135b1b7ed/tools/misc/print_config.py) which can almost be directly applied to MMOCR, but remember to remove `replace_cfg_vals` and `update_data_root` lines to get the origin config.

BTW, would you mind making a PR to help us to add this script back? :)

**Reply by IncludeMathH** [reply]
Well, I'm happy to try to do that.
